tests/brevitas_end_to_end/test_torchvision_models.py

```diff
@@ -113,15 +114,37 @@ def torchvision_model(model_name, quantize_fn):
 
 
 @requires_pt_ge('1.8.1')
-def test_torchvision_graph_quantization_flexml_qcdq_onnx(torchvision_model, request):
+@parametrize('enable_compile', [True, False])
+def test_torchvision_graph_quantization_flexml_qcdq_onnx(
+        torchvision_model, enable_compile, request):
+    test_id = request.node.callspec.id
     if torchvision_model is None:
         pytest.skip('Model not instantiated')
+    if enable_compile:
+        model_name = test_id.split("-")[1]
+        if torch_version <= version.parse('2.2'):
+            pytest.skip("Pytorch 2.2 is required to test compile")
+        else:
+            torch._dynamo.config.capture_scalar_outputs = True
+        if 'vit' in model_name:
+            pytest.skip("QuantMHA not supported with compile")
+
     inp = torch.randn(BATCH, IN_CH, HEIGHT, WIDTH)
-    test_id = request.node.callspec.id
 
     quantize_fn_name = test_id.split("-")[0]
-    torchvision_model(inp)
-    if quantize_fn_name != 'quantize_float':
+    with torch.no_grad(), quant_inference_mode(torchvision_model):
+        prehook_non_compiled_out = torchvision_model(inp)
+        post_hook_non_compiled_out = torchvision_model(inp)
+        assert torch.allclose(prehook_non_compiled_out, post_hook_non_compiled_out)
+
+        if enable_compile:
+            compiled_model = torch.compile(torchvision_model, fullgraph=True)
+            compiled_out = compiled_model(inp)
+
+        # This fails! Compile might needs more small-scoped tests for accuracy evaluation
+        # assert torch.allclose(post_hook_non_compiled_out, compiled_out)
```

Comment:
How badly does it fail? I'd prefer to keep this check with looser bounds if the errors are reasonable...

Reply:
~0.2 absolute error

Reply:
I will add whatever is needed to make the tests pass

Reply:
I added the tolerance and tested locally, but the tests will still not run on the CI because `torch.compile` works only on torch 2.2